api/v1_projects.inc

```diff
@@ -372,6 +372,48 @@ function render_project_json(Project $project, ?array $return_fields = null)
     return $return_array;
 }
 
+//---------------------------------------------------------------------------
+// projects/:projectID/artifacts/:stageid
+
+/** @param array<string,string|string[]> $query_params */
+function api_v1_project_artifacts(string $method, array $data, array $query_params)
+{
+    // get the project and the stage
+    $project = $data[":projectid"];
+    $stage = $data[":stageid"];
+
+    // TODO: This check might be too strict -- do we really care if a user
+    // is able to access artifacts from a stage they can't *work* in?
+    // We only show PP artifacts for users who can access in the stage, but
+    // that doesn't seem strictly necessary.
```

Comment:
Based on a quick look, probably we don't care, unless we want to tighten up on what we allow already. I just checked using one of my alternate TEST IDs that doesn't have PP access. I can't see the PP downloads on a project in the PP pool, but:
* I can see the PP pool page
* I can pull a copy of the CTF with the latest text, which is equivalent to the PP file (except missing the download of the PCs, the GWL and the BWL)
* I can view images online, and download them from the images index.